packages/firestore/src/remote/datastore.ts

```diff
@@ -149,6 +150,33 @@ export async function invokeBatchGetDocumentsRpc(
   return result;
 }
 
+export async function invokeRunQueryRpc(
+  datastore: Datastore,
+  query: Query
+): Promise<Document[]> {
+  const datastoreImpl = debugCast(datastore, DatastoreImpl);
+  const { structuredQuery, parent } = datastoreImpl.serializer.toQueryTarget(
+    query.toTarget()
+  );
+  const params = {
+    database: datastoreImpl.serializer.encodedDatabaseId,
+    parent,
+    structuredQuery
+  };
+
+  const response = await datastoreImpl.invokeStreamingRPC<
+    api.RunQueryRequest,
+    api.RunQueryResponse
+  >('RunQuery', params);
+
+  return (
+    response
+      // Filter RunQueryResponses that only contain readTimes.
```

Comment:
Ultranit: consider s/Filter/Omit/ ("filter" has the unfortunate ambiguity in that it might refer to either the part that is left in place or the part that gets removed. It's clear enough from the context, but I still think something like "omit" is slightly better).

Reply:
Done